test/unit/test.py

```diff
@@ -1835,6 +1836,351 @@ def test_error_template_args(self):
+        
+        data = {i['name']: i.get('value')
+                for i in form.find_all(['input', 'button', 'textarea'])
+                if i.get('value') != 'Leave a Comment'}
+        
+        data[body['name']] = feedback_str
+        
+        save_feedback_path = urlparse(urljoin(url, form['action'])).path
+        response = self.client.post(save_feedback_path, data=data)
+        
+        # View the saved feedback.
+        return self.follow_redirect(response, 303)
+
+    def leave_feedback(self, url, soup, feedback_str):
+        ''' Look for form to leave feedback, submit it and return URL and soup.
+        '''
+        body = soup.find(lambda tag: bool(tag.name == 'textarea' and tag.get('name') == 'comment_text'))
+        form = body.find_parent('form')
+        self.test.assertEqual(form['method'].upper(), 'POST')
+        
+        data = {i['name']: i.get('value')
+                for i in form.find_all(['input', 'button', 'textarea'])
+                if i.get('value') != 'Looks Good!'}
+        
+        data[body['name']] = feedback_str
+        
+        save_feedback_path = urlparse(urljoin(url, form['action'])).path
+        response = self.client.post(save_feedback_path, data=data)
+        
+        # View the saved feedback.
+        return self.follow_redirect(response, 303)
+
+    def approve_activity(self, url, soup):
+        ''' Look for form to approve activity, submit it and return URL and soup.
+        '''
+        body = soup.find(lambda tag: bool(tag.name == 'textarea' and tag.get('name') == 'comment_text'))
+        form = body.find_parent('form')
+        self.test.assertEqual(form['method'].upper(), 'POST')
+        
+        data = {i['name']: i.get('value')
+                for i in form.find_all(['input', 'button', 'textarea'])
+                if i.get('value') != 'Leave a Comment'}
+        
+        approve_activity_path = urlparse(urljoin(url, form['action'])).path
+        response = self.client.post(approve_activity_path, data=data)
+        
+        # View the saved feedback.
+        return self.follow_redirect(response, 303)
+
+    def publish_activity(self, url, soup):
+        ''' Look for form to publish activity, submit it and return URL and soup.
+        '''
+        body = soup.find(lambda tag: bool(tag.name == 'textarea' and tag.get('name') == 'comment_text'))
+        form = body.find_parent('form')
+        self.test.assertEqual(form['method'].upper(), 'POST')
+        
+        data = {i['name']: i.get('value')
+                for i in form.find_all(['input', 'button', 'textarea'])
+                if i.get('value') != 'Leave a Comment'}
+        
+        publish_activity_path = urlparse(urljoin(url, form['action'])).path
+        response = self.client.post(publish_activity_path, data=data)
+        
+        # View the published activity.
+        return self.follow_redirect(response, 303)
+
+class TestApps (TestCase):
```

Comment:
Is there something more descriptive than `TestApps` we could name this? Very close to the existing `TestApp`